bindings/src/main/java/io/pravega/storage/hdfs/HDFSStorage.java

```diff
@@ -454,8 +504,8 @@ public SegmentHandle create(String streamSegmentName) throws StreamSegmentExcept
             throw HDFSExceptionHelpers.convertException(streamSegmentName, HDFSExceptionHelpers.segmentExistsException(streamSegmentName));
         }
 
-        // Create the file for the segment with epoch 0.
-        Path fullPath = getFilePath(streamSegmentName, 0);
+        // Create the file for the segment with current epoch.
```

Comment:
The reason we had epoch 0 here is because this is the only way to avoid two competing instances with different epochs creating the same segment twice. 

If you just create it with the current epoch, then two such competing instances will both be able to create the file so there is a possibility that the cleanup code below fails (system crash) and we are left with extra files, which can spell trouble in the long term.

I see the need for changing it to the current epoch, since we have changed the code in `write` that would fix this file.
I wonder if we could change this to how you did it, but then we should add some code to `openWrite` that automatically cleans up irrelevant files (`findAllRaw` will return that for you), but only if those files are empty.

Reply:
ok. Fixed it by creating file with 0 epoch and then calling claimOwnership which renames files and does other checks and throws StorageNotPrimaryException .  The number of checks should not increase in normal path. In case of race it adds robustnesss. Also now claimOwnership  deletes  old zombie files if present after claiming files.